public/app/features/dashboard-scene/panel-edit/PanelDataPane/TransformationsDrawer.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FormEvent, useCallback, useMemo, useState } from 'react';
+
+import { DataFrame, SelectableValue, standardTransformersRegistry } from '@grafana/data';
+import { IconButton } from '@grafana/ui';
+import { TransformationPickerNg } from 'app/features/dashboard/components/TransformationsEditor/TransformationPickerNg';
+import { FilterCategory } from 'app/features/dashboard/components/TransformationsEditor/TransformationsEditor';
+
+interface DrawerState {
+  search: string;
+  showIllustrations: boolean;
+  selectedFilter?: FilterCategory;
+}
+
+interface TransformationsDrawerProps {
+  series: DataFrame[];
+  isOpen: boolean;
+  onClose: () => void;
+  onTransformationAdd: (selectedItem: SelectableValue<string>) => void;
+}
+
+export function TransformationsDrawer(props: TransformationsDrawerProps) {
+  const { isOpen, series, onClose, onTransformationAdd } = props;
+
+  const [drawerState, setDrawerState] = useState<DrawerState>({
+    search: '',
+    showIllustrations: true,
+  });
+
+  const onSearchChange = useCallback(
+    (e: FormEvent<HTMLInputElement>) => setDrawerState({ ...drawerState, ...{ search: e.currentTarget.value } }),
+    [drawerState]
+  );
+
+  const onShowIllustrationsChange = useCallback(
+    (showIllustrations: boolean): void => setDrawerState({ ...drawerState, ...{ showIllustrations } }),
+    [drawerState]
+  );
+  const onSelectedFilterChange = useCallback(
+    (selectedFilter: FilterCategory): void => setDrawerState({ ...drawerState, ...{ selectedFilter } }),
+    [drawerState]
+  );
+
+  const allTransformations = useMemo(
+    () => standardTransformersRegistry.list().sort((a, b) => (a.name > b.name ? 1 : b.name > a.name ? -1 : 0)),
+    []
+  );
+
+  const transformations = useMemo(
+    () =>
+      allTransformations.filter((t) => {
+        if (
+          drawerState.selectedFilter &&
+          drawerState.selectedFilter !== 'viewAll' &&
+          !t.categories?.has(drawerState.selectedFilter)
+        ) {
+          return false;
+        }
+        return t.name.toLocaleLowerCase().includes(drawerState.search.toLocaleLowerCase());
+      }),
+    [drawerState, allTransformations]
+  );
+
+  const searchBoxSuffix = (
+    <>
+      {transformations.length} / {allTransformations.length} &nbsp;&nbsp;
+      <IconButton
+        name="times"
+        onClick={() => {
+          setDrawerState({ ...drawerState, ...{ search: '' } });
```

Comment:
To have feature parity this should close the drawer if the user clicks the X in the search box, but I found it a bit odd, so I didn't implement it.